simulation/avatar/avatar_wrapper.py

```diff
@@ -60,6 +60,8 @@ def set_code(self, code):
         self.code = code
         try:
             exec(code)
+            if Avatar is None:
+                raise Exception('You need to define a class "Avatar"')
```

Comment:
Ah, sorry, it will bubble to a UserCodeException anyway. Will that display sensibly to the user though, in terms of showing the right message?